### PR TITLE
Cache policy fixes

### PR DESCRIPTION
**Pull request by tonisevener**
Review https://github.com/wikimedia/wikipedia-ios/pull/3478 first.

PR for discussion - added a custom cache policy in  `a9a1365` &  `6b0b008` that is tacked onto the URLRequest header. This gives us the pull to refresh error banner for the user in airplane mode without the previous regression issue of all saved articles showing a failed state.

### CODE REVIEW

File: Wikipedia/Code/Session.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Foundation
 
 public extension URLRequest.CachePolicy {
-    static let noReallyForceRemoteOnly: UInt = 99
+    static let noPersistentCacheOnError: UInt = 99
```
Comment (joewalsh) [inline Wikipedia/Code/Session.swift]: Would it help to have our own `CachePolicy` enum that has a case to wrap an underlying `URLRequest.CachePolicy`?
```
enum CachePolicy {
    case foundation(URLRequest.CachePolicy)
    case noPersistentCacheOnError
}
```

File: Wikipedia/Code/Session.swift [hunk 2]
```diff
@@ -697,3 +699,15 @@ class SessionDelegate: NSObject, URLSessionDelegate, URLSessionDataDelegate {
         callback.success()
     }
 }
+
+fileprivate extension URLRequest {
+    var prefersPersistentCacheOverError: Bool {
+        if let customCachePolicyValue = allHTTPHeaderFields?[URLRequest.customCachePolicyHeaderKey],
```
Comment (joewalsh) [inline Wikipedia/Code/Session.swift]: Might be handy to implement the setter here too

File: Wikipedia/Code/Session.swift (hunk 1 above)
Comment (tonisevener) [reply Wikipedia/Code/Session.swift]: @joewalsh agreed, that feels much cleaner. This is done!

File: Wikipedia/Code/Session.swift (hunk 2 above)
Comment (tonisevener) [reply Wikipedia/Code/Session.swift]: 👍 Done!

**Pull request merged**